frontend/src/components/Wizard/WizardMetadata.vue

```diff
@@ -3,102 +3,108 @@
-      {{ $t('previous') }}
+      :disabled="loading"
+      @click="emit('previous-step')">
+      {{ t('previous') }}
     </v-btn>
     <v-btn
       :loading="loading"
       color="primary"
       variant="elevated"
       @click="setMetadata">
-      {{ $t('next') }}
+      {{ t('next') }}
     </v-btn>
   </div>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue';
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
 import {
   CountryInfo,
   CultureDto,
   StartupConfigurationDto
 } from '@jellyfin/sdk/lib/generated-client';
 import { getLocalizationApi } from '@jellyfin/sdk/lib/utils/api/localization-api';
 import { getStartupApi } from '@jellyfin/sdk/lib/utils/api/startup-api';
-import { useSnackbar } from '@/composables';
+import { useRemote, useSnackbar } from '@/composables';
 
-export default defineComponent({
-  setup() {
-    return {
-      useSnackbar
-    };
-  },
-  data() {
-    return {
-      metadataLanguage: '',
-      metadataCountry: '',
-      initialConfig: {} as StartupConfigurationDto,
-      cultureOptions: [] as CultureDto[],
-      countryOptions: [] as CountryInfo[],
-      loading: false
-    };
-  },
-  async created() {
-    const api = this.$remote.sdk.oneTimeSetup(
-      this.$remote.auth.currentServer?.PublicAddress || ''
-    );
+const emit = defineEmits<{
+  (e: 'step-complete'): void;
+  (e: 'previous-step'): void;
+}>();
 
-    this.initialConfig = (
-      await getStartupApi(api).getStartupConfiguration()
-    ).data;
+const SomeItemSelectedRule = [(v: unknown[]): boolean | string => v.length > 0];
 
-    this.metadataLanguage = this.initialConfig?.MetadataCountryCode || '';
-    this.metadataCountry = this.initialConfig?.PreferredMetadataLanguage || '';
+const { t } = useI18n();
+const remote = useRemote();
 
-    this.cultureOptions = (await getLocalizationApi(api).getCultures()).data;
-    this.countryOptions = (await getLocalizationApi(api).getCountries()).data;
-  },
-  methods: {
-    async setMetadata(): Promise<void> {
-      this.loading = true;
+const metadataLanguage = ref('');
+const metadataCountry = ref('');
+const initialConfig = ref<StartupConfigurationDto>({});
+const cultureOptions = ref<CultureDto[]>([]);
+const countryOptions = ref<CountryInfo[]>([]);
+const loading = ref(false);
 
-      const api = this.$remote.sdk.oneTimeSetup(
-        this.$remote.auth.currentServer?.PublicAddress || ''
-      );
+onMounted(async () => {
+  const api = remote.sdk.oneTimeSetup(
+    remote.auth.currentServer?.PublicAddress ?? ''
+  );
 
-      try {
-        await getStartupApi(api).updateInitialConfiguration({
-          startupConfigurationDto: {
-            ...this.initialConfig,
-            MetadataCountryCode: this.metadataLanguage,
-            PreferredMetadataLanguage: this.metadataCountry
-          }
-        });
+  initialConfig.value = (
+    await getStartupApi(api).getStartupConfiguration()
+  ).data;
```

Comment:
I guess you've been playing this safe since we can't test these component properly because the missing Vuetify components, but I think it makes more sense to fetch all the required data at the page level and just leave the components show it. We've been repeating this request multiple times too.

Reply:
My idea was passing the data as props from the page level, basically

Reply:
I started to make this change, and realized that it will actually regress functionality. Going back to a previous step would not display the data that was previously saved anymore, since the config will only be fetched once at the page level.

eg. Go to Language step, set language and go to the next step. Then go back to the previous step, the language that is auto selected will not be what the user just chose.

The other option is to lift the load & update to the route level, but I think that's just moving code around for not much gain.

Reply:
@aweebs Fair enough!